File: programming files/datetime_convert.cc
#include <iostream>
#include <iomanip>
#include <sstream>
#include <chrono>

int main()
{
    //String containing the date and time of the program
    std::string date_str = "2022-03-17 10:45:30"; 
    //Date object 
    std::tm date_obj = {};
    //Open the input stream and push the date_str to it
    std::istringstream ss(date_str);
    //Use the get time function and initialize the date
    ss >> std::get_time(&date_obj, "%Y-%m-%d %H:%M:%S");
    //Create a formatted date stream
    std::stringstream formatted_date_ss;
    //Create a place to initialize formatted date
    formatted_date_ss << std::put_time(&date_obj, "%m/%d/%Y %H:%M:%S");
    //Convert to string
    std::string formatted_date = formatted_date_ss.str();
    //Print out the formatted date
    std::cout << formatted_date << std::endl;

    return 0;
}

File: programming files/datetime_convert.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
//Create a class for converting date and time
public class datetime_convert {
    //Create main class
    public static void main(String[] args) {
        //Create a string of an initial date
        String dateStr = "2022-03-17 10:45:30";
        //Create formatter object and initialize it with a date format pattern
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        //Create a date object and pass string
        LocalDateTime dateObj = LocalDateTime.parse(dateStr, formatter);
        //Format the object
        String formattedDate = dateObj.format(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss"));
        //Print the date
        System.out.println(formattedDate);
    }
}
